static_frame/core/store_zip.py

```diff
@@ -77,7 +75,17 @@ def _payload_to_frame(cls,
                 name=payload.name,
                 config=payload.config,
                 constructor=payload.constructor,
-        )
+                )
+
+    @staticmethod
+    def _set_container_type(frame: Frame, container_type: tp.Type[Frame]) -> Frame:
+        """
+        Helper method to coerce a frame to the expected type, or return it as is
+        if the type is already correct
+        """
+        if frame.__class__ is not container_type:
+            return frame._to_frame(container_type)
+        return frame
```

Comment:
This is needed explicitly for the case of weakrefs (i.e. there was no existing shortcoming).

Basically, if the user originally requested a `Frame` from the store for label `labelA`, and then later, re-requested a `FrameGO` from the store for the same label (`labelA`), in the old case, it would just read from disk and create a `FrameGO`. With the weak cache though, if the user still has a reference to the original frame, we have to be sure to give back an object of the expected type.